proposal1/README.md

```diff
@@ -0,0 +1,91 @@
+    - Prebuilt [Docker images](https://hub.docker.com/r/ipfs/go-ipfs/) provided,
+      [easy to customize](https://github.com/ipfs/ipfs-docs/pull/1115/files)
+    - Prebuilt [binaries](https://dist.ipfs.io/go-ipfs/) are available too
+  - Supports `--no-copy` import to IPFS
+    ([filestore](https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#ipfs-filestore)
+    or [urlstore](https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#ipfs-urlstore))
+- [pinning service api spec](https://ipfs.github.io/pinning-services-api-spec/)
+  - vendor-agnostic API for asking remote services to pin specific CID and provide ("seed") it to the network
+  - go-ipfs provides a compatible CLI client at `ipfs pin remote --help`
+    - multiple pinning services can be added and used at the same time
+    - pinning can happen in the background, and we can block
+      until N out of M services confirms to have a full copy of the data
+    - `ipfs pin remote add` implements best practices
+      for [provider hints](https://ipfs.github.io/pinning-services-api-spec/#section/Provider-hints)
+      (announcing own multiaddrs as origins and preconnecting to delegate notes from the pinning service)
+- Protocol Labs sponsoring pinning to 2+ services for redundancy and robustness
+    - https://web3.storage
+    - https://estuary.tech
+    - (we most likely can add more)
+
+## Architecture (WIP)
+
+Pipeline at [openzim/zimfarm](https://github.com/openzim/zimfarm) is split into distinct stages.
+
+The [`receiver`](https://github.com/openzim/zimfarm/tree/master/receive) runs
+`zimcheck` on produced ZIM and moves valid ones to `/mnt/quarantine`(?).
+
+We want to perform IPFS import only once per ZIM, and only do it for valid ones.
+
+Need to understand what would be the best integration point for IPFS publishing:
+
+- (X) add it to `receiver` stage (after `zimcheck` but before upload/move)
+- (Y) make it a distinct `ipfs-importer` stage which picks up where `receiver` left,
+  adds .zim to IPFS, and after data is pinned  remotely, adds CID as metadata attribute via `https://api.farm.openzim.org/v1`
+
+I am assuming (Y) is better because it keeps IPFS logic isolated, does not
+impact existing infra, and maybe even allows Protocol Labs to sponsor
+IPFS-related workers, removing any financial burden it could generate for Kiwix.
+
+We don't need to run IPFS node all the time. Cron job could check for new ZIMs
+and perform IPFS import only when a new one is detected. Import would
+start an ephemeral IPFS node with filestore enabled (or urlstore), produce CID
+via `ipfs add --nocopy`, and pin it to remote services.
+
+When we have remote copies confirmed, local IPFS node will be shut down to save
+resources.
+
+### To move forward, need answers for these questions
+
+- should I continue with `./ipfs-importer` and create Dockerfile for running cron-based checker similar to `receiver`, or is there a better way?
```

Comment:
Doing something similar to `receiver` (cron-based checks + directory with zims mounted into Docker container)  allows us to leverage "filestore" from IPFS – do chunking in-memory and point at specific byte ranges of existing files (without duplicating the storage requirement).

Alternative is to run this outside of existing zimfarm and point IPFS node at https://download.kiwix.org/zim/wikipedia/ and use "urlstore". The downside is that IPFS node would have to download entire file twice (once time to do chunking and hashing to produce CID, and second time to send bytes to remote pinning service. 

"filestore" is superior here imo, but mentioning "urlstore" as an option.